Feat/user preference

Emmanuel-Oyewole opened the pull request:
Refactor database migration setup and update Docker configuration

- Changed migration script location from 'migrations' to 'migration' in alembic.ini.
- Updated docker-compose.dev.yml to:
  - Remove env_file and set environment variables directly for PostgreSQL, RabbitMQ, and Redis services.
  - Add health checks for PostgreSQL, Redis, and RabbitMQ services.
  - Define a custom network for the services.
- Added README and env.py for Alembic migrations.
- Created a new script template for Alembic migration scripts.
- Added initial migration for user preferences and related tables.
- Updated pdm.lock and pyproject.toml to include grpcio-tools and adjust dependencies.
- Removed old alembic.ini from src directory.
- Enhanced settings.py to include JWT secret key and other configurations.
- Updated models to include user-related models in the __init__.py.
- Refactored AuthRepository to use AsyncSession and updated methods to be asynchronous.
- Updated AuthService to handle asynchronous calls for user creation and authentication.

Emmanuel-Oyewole committed:
implemented user preference

Emmanuel-Oyewole committed:
started working onn authentication

Emmanuel-Oyewole committed:
Refactor database migration setup and update Docker configuration

- Changed migration script location from 'migrations' to 'migration' in alembic.ini.
- Updated docker-compose.dev.yml to:
  - Remove env_file and set environment variables directly for PostgreSQL, RabbitMQ, and Redis services.
  - Add health checks for PostgreSQL, Redis, and RabbitMQ services.
  - Define a custom network for the services.
- Added README and env.py for Alembic migrations.
- Created a new script template for Alembic migration scripts.
- Added initial migration for user preferences and related tables.
- Updated pdm.lock and pyproject.toml to include grpcio-tools and adjust dependencies.
- Removed old alembic.ini from src directory.
- Enhanced settings.py to include JWT secret key and other configurations.
- Introduced gRPC service structure in src/grpc.
- Updated models to include user-related models in the __init__.py.
- Refactored AuthRepository to use AsyncSession and updated methods to be asynchronous.
- Updated AuthService to handle asynchronous calls for user creation and authentication.

Emmanuel-Oyewole committed:
- fixed import issue in main.py
- changed Sesssion to Asycsession in auth_service dependency

peculiarrichard reviewed (changes requested):
Very extensive work, welldone. 

Please put each table in a different file. Separate the repository/model code as required.  I see everything is in one file. Write tests for your services. Create a root route.py file where all routes will be added, then import it in the main.py file instead of importing all of them in the main.py file directly. I am not seeing any grpc code. Finally, i see that some of your changes have the double quotes while I was able to push with single quotes, I thought we had a linting rule for this? there should be one policy that everyone on the team should follow
Very great job on writing the exceptions util. Thank you so much!

Emmanuel-Oyewole commented:
> Very extensive work, welldone.
> 
> Please put each table in a different file. Separate the repository/model code as required. I see everything is in one file. Write tests for your services. Create a root route.py file where all routes will be added, then import it in the main.py file instead of importing all of them in the main.py file directly. I am not seeing any grpc code. Finally, i see that some of your changes have the double quotes while I was able to push with single quotes, I thought we had a linting rule for this? there should be one policy that everyone on the team should follow Very great job on writing the exceptions util. Thank you so much!

Thanks @peculiarrichard I've written the code for the GRPC side of things, and will be pushing it along with the changes.
Could you help me out with how to enforce linting rule on my IDE? and I've never used pre-commit hooks how do i use and set that up as well?
My question might look silly, apologies

Emmanuel-Oyewole committed:
Refactor user preference management and introduce new models for user consent, notification settings, and privacy settings

- Consolidated user preference routes under a new root route.
- Created separate models for UserConsent, UserNotificationSetting, UserPreference, and UserPrivacySetting.
- Updated the User model to remove embedded preference classes and establish relationships with new models.
- Refactored the preference repository to accommodate new models and methods for managing user preferences, notifications, privacy settings, and consent.
- Enhanced test coverage for the PreferenceService, including tests for getting, updating, and managing user preferences, notification settings, privacy settings, and consent history.
- Introduced constants for consent types to improve code readability and maintainability.
